#163674173 User should be able to create user account

czarjulius opened the pull request:
#### What does this PR do?
It enables the user to register his/her details to the database.

#### Description of Task to be completed?
This will enable a user to store his details in the database
#### How should this be manually tested?
Click register on the index page and enter information.
Then click submit to register.

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#163674173
#### Screenshots or gifs (if appropriate)
#### Questions:

czarjulius committed:
bug(fixedMigrations) : installed @babel-node -ensure all fields captured [Delivers #163713875]

czarjulius committed:
feat(createUser): User can sign up [Delivers #163674173]

ascii-dev commented:
Any reason why your tests and coverage have issues? Thanks.

ascii-dev commented on server/middlewares/validate.js:
```diff
@@ -0,0 +1,66 @@
+    return emailRegex.test(String(email).toLowerCase());
+  },
+};
+export default validation;
```
This could be a class and the functions would be converted to static methods in it. Thanks

ascii-dev commented on server/middlewares/validate.js:
```diff
@@ -0,0 +1,66 @@
+    const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return emailRegex.test(String(email).toLowerCase());
+  },
+};
```
Kindly add a new line after this line. Thanks

ascii-dev commented on server/middlewares/validate.js:
```diff
@@ -0,0 +1,66 @@
+    if (collection.password && !collection.password.replace(/\s/g, '').length) {
+      errors.password = 'Please enter a valid password';
+      isValid = false;
+    }
```
This block of code can be abstracted into a helper function that can be imported here. Kindly look into it. Reach out if you need help.

czarjulius committed:
bug(signUp): Added functions into a single class
- ensure that all comments were cleaned up
- follow the commit message best practice [Delivers #163674173]
